src/main/java/org/opensearch/plugin/insights/core/service/LiveQueriesCache.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.plugin.insights.core.service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.admin.cluster.node.tasks.list.ListTasksRequest;
+import org.opensearch.action.admin.cluster.node.tasks.list.ListTasksResponse;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.tasks.resourcetracker.TaskResourceStats;
+import org.opensearch.core.tasks.resourcetracker.TaskResourceUsage;
+import org.opensearch.plugin.insights.rules.model.Attribute;
+import org.opensearch.plugin.insights.rules.model.Measurement;
+import org.opensearch.plugin.insights.rules.model.MetricType;
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+import org.opensearch.tasks.Task;
+import org.opensearch.tasks.TaskInfo;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+import org.opensearch.transport.client.Client;
+
+/**
+ * Cache for live running queries
+ */
+public class LiveQueriesCache {
+
+    private static final Logger logger = LogManager.getLogger(LiveQueriesCache.class);
+    private static final int MAX_CACHE_SIZE = 1000;
+    private static final int MAX_RETURNED_QUERIES = 100;
+    private static final String SEARCH_ACTION = "indices:data/read/search";
+    private volatile SearchQueryRecord[] sortedQueries = new SearchQueryRecord[0];
+    private final Client client;
+    private final ThreadPool threadPool;
+    private final TransportService transportService;
+    private Scheduler.Cancellable pollingTask;
+
+    public LiveQueriesCache(Client client, ThreadPool threadPool, TransportService transportService) {
+        this.client = client;
+        this.threadPool = threadPool;
+        this.transportService = transportService;
+    }
+
+    public void start() {
+        // Add initial delay to allow cluster initialization, then poll every 10ms
+        pollingTask = threadPool.schedule(this::startPolling, new TimeValue(5, TimeUnit.SECONDS), ThreadPool.Names.GENERIC);
+    }
+
+    private void startPolling() {
+        try {
+            // Check if cluster is ready before starting polling
+            client.admin().cluster().listTasks(new ListTasksRequest(), new ActionListener<ListTasksResponse>() {
+                @Override
+                public void onResponse(ListTasksResponse response) {
```

Comment:
We are ignoring this ListTasksResponse completely. 

Is it even necessary to call `listTasks` inside `startPolling()`? I think we should schedule the recurring job and separately call `pollRunningTasks()` once here or in a parent method.